src/http/ContentLengthInterpreter.h

```diff
@@ -19,12 +21,31 @@ namespace Http
 class ContentLengthInterpreter
 {
 public:
-    explicit ContentLengthInterpreter(const int aDebugLevel);
+    ContentLengthInterpreter();
 
     /// updates history based on the given message-header field
     /// \return true iff the field should be added/remembered for future use
     bool checkField(const String &field);
 
+    /// prohibits Content-Length in 1xx and 204 responses
+    void applyStatusCodeRules(const StatusCode code) {
+        if (!prohibitedAndIgnored_ && ProhibitsContentLength(code))
+            prohibitedAndIgnored_ = (code == scNoContent) ? "prohibited and ignored in the 204 response" :
```

Comment:
Now that these setters are getting complex enough to use data memory allocations (the const strings) they should be moved to the .cc to avoid bloat in the binary from repeated inlining.

Reply:
@yadij, are you sure these string literals create binary bloat? The following C++ [reference](https://en.cppreference.com/w/cpp/language/inline) quote seems to suggest otherwise: `In an inline function ... String literals in all function definitions are shared (they are all the same string literal defined in just one translation unit)`. 

@eduard-bagdasaryan, do you see these strings duplicated in your PR Squid binary? Lack of duplication does not prove that no duplication can happen later, of course, but it is worth a quick check.

FWIW, I could not find any string literal duplication from inlined methods in (unpatched) Squid binaries, but I did not check all literals. I know `text/plain;charset=utf-8`is duplicated, but its (virtual) method is not inlined. I am not against moving these methods into .cc file, but they are otherwise good candidates for inlining, especially the trivial applyTrailerRules().

Reply:
Ah, may be virtuals I am thinking of then - if they are not in fact duplicated in C++11 then I remove my request. That reference "until C++17" is worrying though.

Reply:
Yes, we may need to revisit that for C++17. I hope things are just organized/called differently there because of the introduction of "inlined variables", but perhaps not. We can avoid bloating concerns (while keeping inlining) by moving these string literals into static class members. A bit "heavy" but doable.